girder-dandi-archive/girder_dandi_archive/schema_validation.py

```diff
@@ -0,0 +1,21 @@
+from girder.exceptions import ValidationException
+
+from .constants import SCHEMA_MINIMUM_VERSION
+
+
+def validate_schema(event):
+    meta = event.info["meta"]
+    if "dandiset" not in meta:
+        # Do not validate if there is no dandiset metadata property
+        # Only top level dandiset folders should be validated
+        return
+    dandiset = meta["dandiset"]
+    if "schemaVersion" not in dandiset:
+        raise ValidationException("No schema version specified on folder")
```

Comment:
At first I was thinking we should tell people to upgrade the dandi-cli here, but I think it makes more sense for clients to be able to take whatever action they think is appropriate.

@yarikoptic are these error message sufficient for you?:

>"No schema version specified on folder"

>f"Schema version {schema_version} "
            f"is less than the minimum version {SCHEMA_MINIMUM_VERSION}"

Reply:
```suggestion
        raise ValidationException("dandiset metadata record is missing required schema version. Hint: outdated client")
```
since "folder" is a girder term here since dandiset is stored in the folder.  Also, since unlikely I should parse / match exception messages, a `Hint` would be appreciated by the users.

Reply:
how would `ValidationException` be visible to the client? I mean -- either I could/should parse it out to determine that the action has failed because of `ValidationException` specifically, and then on "what to tell" to the user?  I.e. when should I recommend updating the client if ValidationException has happened. I would prefer not to parse message itself since those could change etc.

Reply:
@yarikoptic to clarify, you want us to add in a message saying "Hint: try to upgrade your dandi-cli" as part of the error message?